klyshko-operator/controllers/tuplegenerationscheduler_controller.go

```diff
@@ -103,10 +124,61 @@ func (r *TupleGenerationSchedulerReconciler) Reconcile(ctx context.Context, req
 	}, nil
 }
 
+// getGeneratorsByTupleType collects available tuple generators into a map indexed by tuple type
+func (r *TupleGenerationSchedulerReconciler) getGeneratorsByTupleType(ctx context.Context) (map[string][]klyshkov1alpha1.TupleGenerator, error) {
+	generators := klyshkov1alpha1.TupleGeneratorList{}
+	err := r.Client.List(ctx, &generators)
+	if err != nil {
+		return nil, fmt.Errorf("fetching available tuple generators failed: %w", err)
+	}
+	generatorsByTupleType := map[string][]klyshkov1alpha1.TupleGenerator{}
+	for _, generator := range generators.Items {
+		for _, support := range generator.Spec.Supports {
+			if generatorsForType, exists := generatorsByTupleType[support.Type]; exists {
+				generatorsByTupleType[support.Type] = append(generatorsForType, generator)
+			} else {
+				generatorsByTupleType[support.Type] = []klyshkov1alpha1.TupleGenerator{generator}
+			}
+		}
+	}
+	return generatorsByTupleType, nil
+}
+
+// getServiceablePolicies filters the policies declared on the given scheduler resource by removing those policies for
+// which no generator is available in the given map of generators
+func (r *TupleGenerationSchedulerReconciler) getServiceablePolicies(
```

Comment:
Update comment to reflect that policies are not considered if more than one generator is available for a given tuple type

Reply:
Done